### ISSUE TITLE
can connect via secure web socket , but could not receive any messages from subscribed channel

### ISSUE THREAD

**Issue by stmkjp** [eclipse/mosquitto#837]
Hi, thanks for the great product!

I had been working with mosquitto without any problem, but suddenly it stops working from a few weeks ago.
React web client (mutt-react) stops responding (webpage using connector stops responding).
First I thought it's react app's problem but it's not. I'm having same results using other clients.

I have configs like below

**mosquitto.conf:**
```
# Place your local configuration in /etc/mosquitto/conf.d/
#
# A full description of the configuration file is at
# /usr/share/doc/mosquitto/examples/mosquitto.conf.example

pid_file /var/run/mosquitto.pid

persistence true
persistence_location /var/lib/mosquitto/

log_dest file /var/log/mosquitto/mosquitto.log

include_dir /etc/mosquitto/conf.d
```

**conf.d/ws.conf:**
```
port 1883

listener 8883
protocol websockets

listener 8083
protocol websockets
certfile /etc/letsencrypt/live/mysite.com/cert.pem
keyfile /etc/letsencrypt/live/mysite.com/privkey.pem
require_certificate false

```

With above config,

1) No problem using default 1883 (tcp) connection (can send/receive messages)
2) can connect to 8883, 8083 (websocket, secure websocket) port, but cannot receive messages ( I guess I managed to send messages, after long delay)
      --> this was tested using MQTTBox osx client which supports websocket/secure websocket connection.
3) using MQTTBox, subscribed websocket connection disconnected after a while (of course no message received even when I send a message using same client)

4) I see socket errors repeatedly in mosquitt.log, I think there's some problem maintaining the websocket connection, but no idea what it is :
```
...
1527490845: Client 0f179cb8-e195-4b9f-9385-8266bdf673811527489970393 has exceeded timeout, disconnecting.
1527490847: Socket error on client 0f179cb8-e195-4b9f-9385-8266bdf673811527489970393, disconnecting.
1527490854: New client connected from xxx.xxx.xxx.xxx as 0f179cb8-e195-4b9f-9385-8266bdf673811527489970393 (c1, k10, u'stmkjp').
1527490880: Socket error on client 0f179cb8-e195-4b9f-9385-8266bdf673811527489970393, disconnecting.
1527490884: New client connected from xxx.xxx.xxx.xxx as 0f179cb8-e195-4b9f-9385-8266bdf673811527489970393 (c1, k10, u'stmkjp').
1527490895: Socket error on client 9325c6fc-9657-4e78-8acb-7b60d5771e8e1527489381718, disconnecting.
...
```

I'm using mosquitto v1.5 from debian repo with libwebsockets 3.0. (server OS: Debian Jessie )
the server is on google cloud compute machine, opened 1883,8083,8883 in/outbound ports

**Could someone help me resolve this?** 
It's really frustrating to have a service not working anymore once was no problem, and don't have any clue...

**Comment by istvank** [comment]
I had the same problem, MQTT.js (browserified) wouldn't want to connect to Mosquitto after an update to v1.5. Fix that worked for me: Update from Jessie to Stretch. :) It updated libwebsockets so I guess that might be a reason for the issue.

**Comment by stmkjp** [comment]
@istvank , thanks for the reply! 
But jessie to stretch....sounds somewhat risky to me.
I might have to fresh install stretch rather than upgrading from jessie, for it's safer (for currently several services are running).

thanks again!

**Comment by istvank** [comment]
I agree that updating the OS for Mosquitto v1.5 to work is a huge task. IMHO Mosquitto should still work under Jessie, there's another issue related to that here #825 and I guess more will pop up when people update their packages on Raspbian Jessie. Should this issue be reopened or should we create another one..?

**Comment by stmkjp** [comment]
OK, I'll reopen this thread to support resolving the mosquitto / libwebsockets issue on debian jessie.
Meanwhile, I decided to use mosca temporarily as an alternative for mosquitto for a quick remedy.

Fingers crossed for this issue can be resolved soon.

**Comment by timoline** [comment]
1.5 indeed broke websockets on debian jessie...

Can you please fix this ?

**Comment by timoline** [comment]
Maybe its because. in 1.5...
Minimum supported libwebsockets version is now 1.3.

**Comment by Tifaifai** [comment]
I invite you to install libwebsocket >=2.4.1 and close your issue ;)
https://github.com/eclipse/mosquitto/issues/303#issuecomment-364653575
C-U

**Comment by stmkjp** [comment]
@Tifaifai , thank you for the suggestion, but currently I’m having the issue with libwebsocket 3.0, which is newer than 2.4.1 as you can see on my first post.

**Comment by Tifaifai** [comment]
have you testing with downgrade lws 3.0 to 2.4.2 or 2.4.1 ??

**Comment by stmkjp** [comment]
@Tifaifai , this issue happened after upgrading, so it can be resolved if i downgrade related packages. But IMHO, it’s not a preferred way of resolving issues. I’m using Mosca as an alternative for now with no problem, while re-opening this issue hoping that this issue can be referenced by next commit. 

But anyway, i agree that downgrading  can be another way of temporary fix. thanks for the tip.

**Comment by timoline** [comment]
Like above, it should work out of the box, not with some work-arounds

**Comment by Razikus** [comment]
Hi, i know that's old, but this is probably related to "keep_alive" client configuration
Try to set lower one

I've spent on this some thime, and checking every configuration, and after changing keep alive to 10 seconds it was working.

**Comment by ralight** [comment]
The versions involved are now obsolete, so I'm closing this issue.
